models/net.py

```diff
@@ -253,78 +221,65 @@ def forward(
 
         # step 2. Learning-based patchmatch
         depth = torch.empty(0)
+        depths: List[torch.Tensor] = []
+        score = torch.empty(0)
         view_weights = torch.empty(0)
-        depth_patchmatch: Dict[str, List[torch.Tensor]] = {}
-
-        # Unroll this loop since TorchScript only allows "getattr" access with string literals
-        # Stage 3
-        src_features_l = [src_fea["stage_3"] for src_fea in src_features]
-        ref_proj, src_projs = proj_matrices_3[0], proj_matrices_3[1:]
-        depth, _, view_weights = self.patchmatch_3(
-            ref_feature=ref_feature["stage_3"],
-            src_features=src_features_l,
-            ref_proj=ref_proj,
-            src_projs=src_projs,
-            depth_min=depth_min,
-            depth_max=depth_max,
-            depth=depth,
-            img=imgs_3_ref,
-            view_weights=view_weights,
-        )
-        depth_patchmatch["stage_3"] = depth
-        depth = depth[-1].detach()
-        # upsampling the depth map and pixel-wise view weight for next stage
-        depth = F.interpolate(depth, scale_factor=2.0, mode="nearest")
-        view_weights = F.interpolate(view_weights, scale_factor=2.0, mode="nearest")
-
-        # Stage 2
-        src_features_l = [src_fea["stage_2"] for src_fea in src_features]
-        ref_proj, src_projs = proj_matrices_2[0], proj_matrices_2[1:]
-        depth, _, view_weights = self.patchmatch_2(
-            ref_feature=ref_feature["stage_2"],
-            src_features=src_features_l,
-            ref_proj=ref_proj,
-            src_projs=src_projs,
-            depth_min=depth_min,
-            depth_max=depth_max,
-            depth=depth,
-            img=imgs_2_ref,
-            view_weights=view_weights,
-        )
-        depth_patchmatch["stage_2"] = depth
-        depth = depth[-1].detach()
-        # upsampling the depth map and pixel-wise view weight for next stage
-        depth = F.interpolate(depth, scale_factor=2.0, mode="nearest")
-        view_weights = F.interpolate(view_weights, scale_factor=2.0, mode="nearest")
-
-        # Stage 1
-        src_features_l = [src_fea["stage_1"] for src_fea in src_features]
-        ref_proj, src_projs = proj_matrices_1[0], proj_matrices_1[1:]
-        depth, score, _ = self.patchmatch_1(
-            ref_feature=ref_feature["stage_1"],
-            src_features=src_features_l,
-            ref_proj=ref_proj,
-            src_projs=src_projs,
-            depth_min=depth_min,
-            depth_max=depth_max,
-            depth=depth,
-            img=imgs_1_ref,
-            view_weights=view_weights,
-        )
-        depth_patchmatch["stage_1"] = depth
-        depth = depth[-1].detach()
+        depth_patchmatch: Dict[int, List[torch.Tensor]] = {}
+
+        for stage in range(self.stages - 1, 0, -1):

```

Comment:
This can be now a loop, but it still needs the conditionals inside to satisfy TorchScript constraints.